uwblueprint/sistering
# Add volunteers and employees to seed data

## Implementation description
*  There are now two employees and two volunteers, each assigned to different branches.
*  Each posting is assigned to a different employee
* Added some signups


## Steps to test
Run `vault kv get -format=json kv/sistering | python update_secret_files.py` to get the newest UIDs for the new test users.
You may need to update your vault credentials which can be found [here](https://discord.com/channels/744991813248745522/883741333741776976/932377555128959007).

The login credentials are:
Admin: sistering+admin@uwblueprint.org
Volunteer: sistering+volunteer@uwblueprint.org
Volunteer1: sistering+volunteer1@uwblueprint.org
Employee: sistering+employee@uwblueprint.org
Employee1: sistering+employee1@uwblueprint.org

Password is Test123 for all accounts.

Run `docker exec -it sistering_backend bash -c "npx prisma migrate reset"` to remove the all the data you currently have, with the seed data. If you don't want to delete your current data, you can run `npx prisma db seed` instead. 

Check that the users, volunteers, employees, postings, shifts, signups, skills, and branches were created properly. This can be done through graphql or postgres. Some useful graphql queries:
```
query Postings {
  postings {
    id
    title
    description
    startDate
    endDate
    status
    autoClosingDate
    branch {
      id
    }
    skills {
      name
    }
    shifts {
      startTime
      endTime
    }
  }
}

query Users {
  users {
    id
    email
    firstName
    lastName
  }
}

query Volunteers {
  volunteerUsers {
    id
    lastName
    firstName
    dateOfBirth
    hireDate
    branches {
      id
      name
    }
  }
}

... etc
```

Login as an admin:
1. Go to http://localhost:3000/admin/schedule/posting/1
2. Check that the volunteer with the correct branch is signed up for some shifts

Login as Volunteer:
1. Go to `http://localhost:3000/volunteer/postings` and check that only postings from the branch `Arts` are displayed
2. Go to `http://localhost:3000/volunteer/posting/1/availabilities` and check that the correct shifts are displayed in the table (they don't need to be checkmarked)

Repeat above for Volunteer1 for the `Kitchen` branch.


## What should reviewers focus on?
* The data was created properly
* Data is rendering properly in the UI


## Checklist
